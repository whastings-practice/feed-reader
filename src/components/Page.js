import React from 'react';
import Link from 'redux-first-router-link';
import { Container, Header, Menu } from 'semantic-ui-react';

export default function Page(props) {
  return (
    <div>
      <header>
        <Menu size="massive" borderless>
          <Container>
            <Header as="h1">
              <Link to="/">Read &apos;Dem Feeds!</Link>
            </Header>
          </Container>
        </Menu>
      </header>
      <Container>
        {props.children}
      </Container>
    </div>
  );
}
