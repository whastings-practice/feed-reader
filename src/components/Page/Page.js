import React from 'react';
import PropTypes from 'prop-types';
import Link from 'redux-first-router-link';
import Container from 'reactstrap/lib/Container';
import Row from 'reactstrap/lib/Row';
import Col from 'reactstrap/lib/Col';
import Navbar from 'reactstrap/lib/Navbar';
import NavbarBrand from 'reactstrap/lib/NavbarBrand';

import styles from './Page.module.css';
import FeedsContainer from '../../containers/Feeds/FeedsContainer';

const propTypes = {
  children: PropTypes.node.isRequired,
};

export default function Page(props) {
  return (
    <div>
      <Navbar className={styles.navContainer}>
        <Container>
          <NavbarBrand tag="h1">
            <Link to="/" className={styles.brandLink}>Read &apos;Dem Feeds!</Link>
          </NavbarBrand>
        </Container>
      </Navbar>
      <Container>
        <Row>
          <Col xs="12" sm="3">
            <FeedsContainer />
          </Col>
          <Col xs="12" sm="9">
            {props.children}
          </Col>
        </Row>
      </Container>
    </div>
  );
}

Page.propTypes = propTypes;
