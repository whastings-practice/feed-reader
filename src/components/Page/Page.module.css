.navContainer {
  background-color: #007fab;
  border-color: #0f6181;
  border-style: solid;
  border-width: 0 1px 4px 1px;
  margin-bottom: 20px;
}

.brandLink {
  color: #fff;
}
