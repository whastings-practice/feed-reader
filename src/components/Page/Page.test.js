import React from 'react';
import { shallow } from 'enzyme';

import Page from './Page';

describe('<Page />', () => {
  let wrapper;

  function render() {
    wrapper = shallow(
      <Page>
        <div>Child</div>
      </Page>
    );
  }

  beforeEach(() => {
    render();
  });

  test('it renders site nav, feeds list, and any children', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
