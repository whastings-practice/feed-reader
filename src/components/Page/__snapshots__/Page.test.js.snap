// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> it renders site nav, feeds list, and any children 1`] = `
<div>
  <Navbar
    tag="nav"
    toggleable={false}
  >
    <Container
      tag="div"
    >
      <NavbarBrand
        tag="h1"
      >
        <Connect(Link)
          to="/"
        >
          Read 'Dem Feeds!
        </Connect(Link)>
      </NavbarBrand>
    </Container>
  </Navbar>
  <Container
    tag="div"
  >
    <Row
      tag="div"
    >
      <Col
        sm="3"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <Connect(FeedsList) />
      </Col>
      <Col
        sm="9"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <div>
          Child
        </div>
      </Col>
    </Row>
  </Container>
</div>
`;
